pkg/secp256k1/keypair.go

```diff
@@ -43,11 +43,17 @@ func GenerateSecp256k1KeyPair() (*KeyPair, error) {
 	return wrapSecp256k1Key(key, key.PubKey()), nil
 }
 
+// Note there is no error condition returned by this function (use KeyPairFromBytes)
```

Comment:
Should we prefix this with `Deprecated:`? I think some Go docs will pick up on that if we want to strongly suggest switching to `KeyPairFromBytes`.